beancount_import/source/schwab_csv.py

```diff
@@ -598,6 +610,74 @@ def get_narration_prefix(self) -> str:
         return "TRANSFER"
 
 
+@dataclass(frozen=True)
+class StockSplit(TransactionEntry):
+    capital_gains_account: str
+    lot_splits: List[LotSplit]
+
+    def get_sub_account(self) -> Optional[str]:
+        return self.amount.currency
+
+    def get_postings(self) -> List[Posting]:
+        postings = []
+        if not self.lot_splits:
+            return super().get_postings()
+
+        for split in self.lot_splits:
+            postings.append(
+                Posting(
+                    account=self.get_primary_account(),
+                    units=Amount(-split.prev_qty, self.amount.currency),
+                    cost=CostSpec(
+                        number_per=split.prev_cost,
+                        number_total=None,
+                        currency=CASH_CURRENCY,
+                        date=split.date,
+                        label=None,
+                        merge=None,
+                    ),
+                    price=None,
+                    flag=None,
+                    meta=self.get_meta(),
+                )
+            )
+            postings.append(
+                Posting(
+                    account=self.get_primary_account(),
+                    units=Amount(split.new_qty, self.amount.currency),
+                    cost=CostSpec(
+                        number_per=split.new_cost,
+                        number_total=None,
+                        currency=CASH_CURRENCY,
+                        date=split.date,
+                        label=None,
+                        merge=None,
+                    ),
+                    price=None,
+                    flag=None,
+                    meta=self.get_meta(),
+                )
+            )
+
+        postings.append(
```

Comment:
Do splits ever involve some forced selling?
If the arithmetic is correct, there should be no need for any other posting.

Reply:
You're right, and good catch! Will remove this. I think in practice I had a need for this posting in one split due to inexact cost figures, so a penny or two needed absorbing. But I don't think with this automation that should occur, and it's better to make the error explicit if it does.